backends-velox/src/main/scala/org/apache/spark/sql/execution/ColumnarCachedBatchSerializer.scala

```diff
@@ -261,7 +261,11 @@ class ColumnarCachedBatchSerializer extends CachedBatchSerializer with SQLConfHe
                 .deserialize(deserializerHandle, cachedBatch.bytes)
             val batch = ColumnarBatches.create(ExecutionCtxs.contextInstance(), batchHandle)
             if (shouldPruning) {
-              ColumnarBatches.select(nmm, batch, requestedColumnIndices.toArray)
+              try {
+                ColumnarBatches.select(nmm, batch, requestedColumnIndices.toArray)
+              } finally {
+                batch.close()
```

Comment:
the batch is released at native side `select`, do we still need close it at jvm side ?

Reply:
I see. I'd remove the release call from native code then let user to close the input batch manually. A batch-handle is usually mapped to a Java-side `IndicatorVector` which has a ref-cnt to manage the underlying buffers. Calling release from native code would break this kind of memory management.